Content.Server/Radio/EntitySystems/JammerSystem.cs

```diff
@@ -102,12 +139,64 @@ private bool ShouldCancelSend(EntityUid sourceUid)
 
         while (query.MoveNext(out _, out _, out var jam, out var transform))
         {
-            if (source.InRange(EntityManager, _transform, transform.Coordinates, jam.Range))
+            if (source.InRange(EntityManager, _transform, transform.Coordinates, GetCurrentRange(jam)))
             {
                 return true;
             }
         }
 
         return false;
     }
+
+    private void OnGetVerb(Entity<RadioJammerComponent> entity, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        var user = args.User;
+
+        byte index = 0;
+        foreach (var setting in entity.Comp.Settings)
+        {
+            // This is because Act wont work with index.
+            // Needs it to be saved in the loop.
+            var currIndex = index;
+            var verb = new Verb
+            {
+                Priority = currIndex,
+                Category = VerbCategory.PowerLevel,
+                Disabled = entity.Comp.SelectedPowerLevel == currIndex,
+                Act = () =>
+                {
+                    entity.Comp.SelectedPowerLevel = currIndex;
+                    var jammerComp = Comp<DeviceNetworkJammerComponent>(entity.Owner);
+                    jammerComp.Range = GetCurrentRange(entity.Comp);
+                    jammerComp.Dirty();
```

Comment:
```suggestion
                    if(TryComp<DeviceNetworkJammerComponent>(entity.Owner, out var jammerComp))
                    {
                        jammerComp.Range = GetCurrentRange(entity.Comp);
                        Dirty(entity.Owner, jammerComp);
                    }
```

Calling Comp will cause an exception if the component does not exist.

Calling dirty on components is deprecated. You should use Dirty(uid, component)

Reply:
This is partly my fault, for some reason there was a space between Try and Comp on my original comment.

Reply:
Haha that's a funny misunderstanding 😆! I totally thought you meant to try using Comp. Making this change now!

Reply:
By the way, you are able to mark reviews as resolved and you should do they don't show up as active for anyone else viewing.